### DISCUSSION TITLE
Vivo Brazil second region - Região 2

### DISCUSSION THREAD

**Discussion by jotala** [fabianishere/udm-iptv#188] [Q&A]
I’ve been trying to set up IPTV for more than a year.
I read every discussion and issue, and I thought it was a DHCP option 43 and 121 problems. #96 
It is my first time using GitHub, excuse me for any mistakes. 

I think I found out:

Maybe  @apedroheringer setup uses an old IPTV system from Vivo ISP, called Media Room, that’s why his setup is working.
Like @cawv my system is Open Platform.

On the ISP router, it shows on the WAN interfaces, and it VLANs

600 Internet
601 VoIP
602 IPTV
4000 Multicast

(Mine VLAN is different from the others, because Vivo differentiated VLAN based on regions, this is not an issue as successfully got the routes and IP on VLAN 602)

What I saw on Wireshark:
IPTV goes thru the IP gathered from DHCP on 602
Multicast origins are from 10.0.0.1 on VLAN 4000

I don’t know if is the same problem from “PostTV” #86 where the guy is using “eth8.35” who needed the IPTV originates from 10.10.10.10

With the current configuration, I was able to get the IP and routes, the Live TV works for about 10 seconds then freezes, if I change the channel the behavior repeats. 
I can watch and use the DVR function to rewind a show and it never freezes as long I’m behind the live streaming. (I think this is due to unicast working properly)

@fabianishere is it possible to have two Vlans on the IPTV interface, or segregate the multicast VLAN?

If you want I could share the file from Wireshark, also the routes, etc.

```
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 602 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 0.0.0.0/0
LAN Interfaces: br6
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
73: iptv@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.182.A.BBB/20 scope global iptv
       valid_lft forever preferred_lft forever
10.182.0.0/20 proto kernel scope link src 10.182.A.BBB 
172.28.0.0/16 via 10.182.0.1 metric 273 
177.16.30.0/23 via 10.182.0.1 metric 275 
200.161.71.40/30 via 10.182.0.1 metric 276 
200.161.71.46/31 via 10.182.0.1 metric 277 
200.161.71.48/31 via 10.182.0.1 metric 278 
201.0.52.0/23 via 10.182.0.1 metric 274 
=== Service Logs ===
Sep 06 15:24:03 ubnt udm-iptv[9091]: Inserted route table entry for 239.130.2.0 on VIF #0
Sep 06 15:24:03 ubnt udm-iptv[9091]: Joining group 239.130.2.0 on interface iptv
Sep 06 15:24:04 ubnt udm-iptv[9091]: RECV V2 member report   from 192.168.15.244  to 239.131.0.0
Sep 06 15:24:04 ubnt udm-iptv[9091]: Updated route entry for 239.131.0.0 on VIF #0
Sep 06 15:24:04 ubnt udm-iptv[9091]: RECV Leave message      from 192.168.15.244  to 224.0.0.2
Sep 06 15:24:04 ubnt udm-iptv[9091]: Leaving group 239.131.0.0 on interface iptv
Sep 06 15:24:04 ubnt udm-iptv[9091]: RECV Membership query   from 192.168.15.1    to 239.131.0.0
Sep 06 15:24:04 ubnt udm-iptv[9091]: RECV V2 member report   from 192.168.15.244  to 239.131.0.0
Sep 06 15:24:04 ubnt udm-iptv[9091]: Updated route entry for 239.131.0.0 on VIF #0
Sep 06 15:24:04 ubnt udm-iptv[9091]: Joining group 239.131.0.0 on interface iptv
```

_Originally posted by @jotala in https://github.com/fabianishere/udm-iptv/discussions/152#discussioncomment-3578531_

**Comment by jotala** [comment]
I have made advance @fabianishere 
I'm so happy!

I managed to make progress on this problem.
Vivo ISP has two configurations, this is number 2.

Confirmed
VLAN 602 - VOD (On demand)
VLAN 4000 - IGMP Multicast

I managed to make it work by multicast by changing the following settings

```
IPTV_WAN_INTERFACE="eth8"
IPTV_WAN_VLAN="4000"
IPTV_WAN_VLAN_INTERFACE="iptv"
IPTV_WAN_RANGES="0.0.0.0/0"
IPTV_LAN_INTERFACES="br602"
IPTV_IGMPPROXY_DISABLE_QUICKLEAVE="false"
IPTV_IGMPPROXY_DEBUG="true"
IPTV_WAN_DHCP="false"
IPTV_WAN_STATIC_IP="10.0.0.1/32"
```


What I need now is to configure a new eth.602 Interface for the VOD and its routes.


**IGMP:**
VLAN ID - 4000
WAN STATIC IP= 10.0.0.1/32
no DHCP

**Video on demand**
The STB guide and update comes from this Interface
VLAN - 602
Wan Range: 172.28.0.0/14 177.16.0.0/16 200.161.71.0/24 201.0.52.0/23

I tried to figure out how to do it myself by reading from this forum:
https://gathering.tweakers.net/forum/list_messages/2065234/2 and this is where I got:

```
IP link add link eth8 name iptvvod type vlan id 602
udhcpc -d /bin/true -s /bin/true -i iptvvod
iptables -t nat -A POSTROUTING -j MASQUERADE -o iptvvod
ip link set iptvvod up

iptables -t nat -A POSTOUTING -d 172.28.0.0/14 -j MASQUERADE -o iptvvod
iptables -t nat -A POSTOUTING -d 177.16.0.0/16 -j MASQUERADE -o iptvvod
iptables -t nat -A POSTOUTING -d 200.161.71.0/24 -j MASQUERADE -o iptvvod
iptables -t nat -A POSTOUTING -d 201.0.52.0/23 -j MASQUERADE -o iptvvod 
```

Can you please help me set up this interface to come back on reboot and add the necessary routes?

**Comment by apedroheringer** [comment]
> I have made advance @fabianishere 
> I'm so happy!
> 
> I managed to make progress on this problem.
> Vivo ISP has two configurations, this is number 2.
> 
> Confirmed
> VLAN 602 - VOD (On demand)
> VLAN 4000 - IGMP Multicast
> 
> I managed to make it work by multicast by changing the following settings
> 
> ```
> IPTV_WAN_INTERFACE="eth8"
> IPTV_WAN_VLAN="4000"
> IPTV_WAN_VLAN_INTERFACE="iptv"
> IPTV_WAN_RANGES="0.0.0.0/0"
> IPTV_LAN_INTERFACES="br602"
> IPTV_IGMPPROXY_DISABLE_QUICKLEAVE="false"
> IPTV_IGMPPROXY_DEBUG="true"
> IPTV_WAN_DHCP="false"
> IPTV_WAN_STATIC_IP="10.0.0.1/32"
> ```
> 
> 
> What I need now is to configure a new eth.602 Interface for the VOD and its routes.
> 
> 
> **IGMP:**
> VLAN ID - 4000
> WAN STATIC IP= 10.0.0.1/32
> no DHCP
> 
> **Video on demand**
> The STB guide and update comes from this Interface
> VLAN - 602
> Wan Range: 172.28.0.0/14 177.16.0.0/16 200.161.71.0/24 201.0.52.0/23
> 
> I tried to figure out how to do it myself by reading from this forum:
> https://gathering.tweakers.net/forum/list_messages/2065234/2 and this is where I got:
> 
> ```
> IP link add link eth8 name iptvvod type vlan id 602
> udhcpc -d /bin/true -s /bin/true -i iptvvod
> iptables -t nat -A POSTROUTING -j MASQUERADE -o iptvvod
> ip link set iptvvod up
> 
> iptables -t nat -A POSTOUTING -d 172.28.0.0/14 -j MASQUERADE -o iptvvod
> iptables -t nat -A POSTOUTING -d 177.16.0.0/16 -j MASQUERADE -o iptvvod
> iptables -t nat -A POSTOUTING -d 200.161.71.0/24 -j MASQUERADE -o iptvvod
> iptables -t nat -A POSTOUTING -d 201.0.52.0/23 -j MASQUERADE -o iptvvod 
> ```
> 
> Can you please help me set up this interface to come back on reboot and add the necessary routes?

Do you have this on your UDM?
https://github.com/unifi-utilities/unifios-utilities
It's really simple with on-boot-script
